bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -359,8 +379,21 @@ public void runWithFlags(boolean force, boolean suspendMajor, boolean suspendMin
             minorCompacting.set(true);
             doCompactEntryLogs(minorCompactionThreshold);
             lastMinorCompactionTime = System.currentTimeMillis();
+            if (!enableMajorCompaction || suspendMajor) {
+                /*
+                 * if major compaction is disabled or if it is suspended, then
+                 * consider minor compaction as main compaction.
+                 */
+                lastMainCompactionTime.set(lastMinorCompactionTime);
+            }
             gcStats.getMinorCompactionCounter().inc();
             minorCompacting.set(false);
+        } else if (((!enableMajorCompaction) || suspendMajor) && ((!enableMinorCompaction) || suspendMinor)) {
+            /*
+             * if both compactions (major and minor) are disabled / suspended,
+             * then consider regular gc run as main compaction
+             */
+            lastMainCompactionTime.set(curTime);
```

Comment:
What is the downside of any major/minor/gc as last compaction time? This  to make a decision if we need to run force GC or not right?

Reply:
I dont want to delay forcegc, if major compaction is enabled and just minor compaction has run recently. Because, if forcegc is triggered immediately without delay, then there is good chance of reclaiming enough space to become writable again.